number_search.py

```diff
@@ -4,24 +4,38 @@
 import re
 
 def count_numbers(text):
-    """
-        counts whole number and decimals i.e -ve and +ve whole numbers decimals
-        TODO update doc strings
-    """
-    pass #TODO update the fucntion to pass test
+    pattern = r"(?:-?\\frac{\d+}{\d+}|-?\d+\\tinyfrac{\d+}{\d+}|[A-Z]+[0-9]+|[a-z][0-9]+|-?\d+\.? ?,?\d+|-?\d+)"
 
+    with open(text, "r") as f:
+        text = f.read()
+        regex = re.compile(char)
+        matched = regex.findall(text)
```

Comment:
any number preceded by anything other than a comma or full stop is not a valid... using the findall will match all the numbers because you did'nt account for it in the pattern.. i suggest you use .split() method